File: src/tools/pmd-cpd/eval-repos/synthetic/go/light_duplication.go
// Package synthetic contains Go files for CPD testing with light duplication.
package synthetic

import (
	"errors"
	"fmt"
	"strings"
)

// UserData represents user data.
type UserData struct {
	ID          int
	Name        string
	Email       string
	Active      bool
	CreatedAt   string
	Permissions []string
}

// ProcessUserData processes user data with validation.
func ProcessUserData(user UserData) UserData {
	return UserData{
		ID:        user.ID,
		Name:      strings.TrimSpace(user.Name),
		Email:     strings.ToLower(strings.TrimSpace(user.Email)),
		Active:    user.Active,
		CreatedAt: user.CreatedAt,
	}
}

// ProcessAdminData processes admin data with validation - light duplication.
func ProcessAdminData(admin UserData) UserData {
	return UserData{
		ID:          admin.ID,
		Name:        strings.TrimSpace(admin.Name),
		Email:       strings.ToLower(strings.TrimSpace(admin.Email)),
		Active:      admin.Active,
		CreatedAt:   admin.CreatedAt,
		Permissions: admin.Permissions,
	}
}

// ValidateEmail validates email format.
func ValidateEmail(email string) bool {
	if email == "" {
		return false
	}
	if !strings.Contains(email, "@") {
		return false
	}
	parts := strings.Split(email, "@")
	if len(parts) != 2 {
		return false
	}
	if parts[0] == "" || parts[1] == "" {
		return false
	}
	if !strings.Contains(parts[1], ".") {
		return false
	}
	return true
}

// FormatCurrency formats a number as currency.
func FormatCurrency(amount float64, currency string) string {
	symbols := map[string]string{
		"USD": "$",
		"EUR": "E",
		"GBP": "P",
	}
	symbol, ok := symbols[currency]
	if !ok {
		symbol = currency
	}
	if amount < 0 {
		return fmt.Sprintf("-%s%.2f", symbol, -amount)
	}
	return fmt.Sprintf("%s%.2f", symbol, amount)
}

// CalculateDiscount calculates discounted price.
func CalculateDiscount(price, discountPct float64) (float64, error) {
	if discountPct < 0 || discountPct > 100 {
		return 0, errors.New("discount must be between 0 and 100")
	}
	return price * (1 - discountPct/100), nil
}

File: src/tools/pmd-cpd/eval-repos/synthetic/java/LightDuplication.java
/**
 * Java file with light duplication - small duplicated blocks.
 */
package synthetic;

import java.util.*;

public class LightDuplication {

    public static Map<String, Object> processUserData(Map<String, Object> user) {
        Map<String, Object> result = new HashMap<>();
        result.put("id", user.getOrDefault("id", 0));
        result.put("name", ((String) user.getOrDefault("name", "")).trim());
        result.put("email", ((String) user.getOrDefault("email", "")).toLowerCase().trim());
        result.put("active", user.getOrDefault("active", true));
        result.put("createdAt", user.getOrDefault("createdAt", ""));
        return result;
    }

    public static Map<String, Object> processAdminData(Map<String, Object> admin) {
        Map<String, Object> result = new HashMap<>();
        result.put("id", admin.getOrDefault("id", 0));
        result.put("name", ((String) admin.getOrDefault("name", "")).trim());
        result.put("email", ((String) admin.getOrDefault("email", "")).toLowerCase().trim());
        result.put("active", admin.getOrDefault("active", true));
        result.put("createdAt", admin.getOrDefault("createdAt", ""));
        result.put("permissions", admin.getOrDefault("permissions", Collections.emptyList()));
        return result;
    }

    public static boolean validateEmail(String email) {
        if (email == null || email.isEmpty()) return false;
        if (!email.contains("@")) return false;
        String[] parts = email.split("@");
        if (parts.length != 2) return false;
        if (parts[0].isEmpty() || parts[1].isEmpty()) return false;
        if (!parts[1].contains(".")) return false;
        return true;
    }

    public static String formatCurrency(double amount, String currency) {
        Map<String, String> symbols = Map.of("USD", "$", "EUR", "E", "GBP", "P");
        String symbol = symbols.getOrDefault(currency, currency);
        if (amount < 0) {
            return String.format("-%s%.2f", symbol, Math.abs(amount));
        }
        return String.format("%s%.2f", symbol, amount);
    }

    public static double calculateDiscount(double price, double discountPct) {
        if (discountPct < 0 || discountPct > 100) {
            throw new IllegalArgumentException("Discount must be between 0 and 100");
        }
        return price * (1 - discountPct / 100);
    }
}
